meson.build

```diff
@@ -64,9 +64,19 @@ warning_c_flags = [
   '-Wno-error=varargs'
 ]
 
+arch = host_machine.cpu_family()
+
+if get_option('enable-avx')
```

Comment:
Please don't enable-avx if arch is not x64. We need to emit erros or force-disable avx if it's not x64 in meson  script.

You may use tri-state (meson feature) instead of bool for the convinience of avx option.
Currenty, you are enabling avx in Ubuntu-arm/risc-v.